libzmix/src/verifiable_encryption/cs_verifiable_encryption.rs

```diff
@@ -0,0 +1,849 @@
+// Copyright contributors to Hyperledger Ursa.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Camenisch-Shoup verifiable encryption.
+/// Based on the paper Practical Verifiable Encryption and Decryption of Discrete Logarithms. https://www.shoup.net/papers/verenc.pdf
+/// Need to be used with Anonymous credentials as described in Specification of the Identity
+/// Mixer Cryptographic Library, https://domino.research.ibm.com/library/cyberdig.nsf/papers/EEB54FF3B91C1D648525759B004FBBB1/$File/rz3730_revised.pdf
+/// sections 5.3, 6.2.10 and 6.2.19
+use super::bn::{BigNumber, BigNumberContext, BIGNUMBER_1, BIGNUMBER_2};
+use super::errors::prelude::*;
+
+use super::cl::constants::*;
+use super::cl::hash::get_hash_as_int;
+use super::cl::helpers::*;
+
+#[derive(Serialize, Deserialize)]
+pub struct PaillierGroup {
```

Comment:
Can we use better names than single letters? Like 'generator'. 

Also, can you reference wherein the paper these values are coming from? Page numbers at least.

Reply:
The symbols `g` and `h` are chosen intentionally to map to the paper. The relevant sections like `Keypair` and others specify which sections of the paper they are taken from and `PaillierGroup` is created in `Keypair`.

Reply:
You can still make comments that state this references `g` or `h` from the paper.

Reply:
Ok

Reply:
Done